omnigibson/object_states/temperature.py

```diff
@@ -93,6 +74,9 @@ def _update(self):
 
         self.value = new_temperature
 
+        # Also update cache to force syncing of temperature so get_value() is consistent with self.value
+        self.update_cache(get_value_args=())
```

Comment:
hmm why is this needed? seems fairly hacky.

Reply:
Remove this. Also remove circular import in HeatSourceOrSink, and re-implement `compute_temperature_delta(self, obj)` as `affect_object(self, obj)`, which return True/False.